### DISCUSSION TITLE
trust 0%

### DISCUSSION THREAD

**Discussion by 190031994Harsha** [abrahamjuliot/creepjs#251] [General]
I am getting a trust score of 0%. I am a total beginner can someone tell what is this and how can I increase my trust score to the maximum. I am just scarping this website casually and making a report for educational purpose

**Comment by 190031994Harsha** [comment]
@abrahamjuliot what are all these sir? where can i learn about these
Navigator.deviceMemory:
- failed descriptor.value undefined

Navigator.hardwareConcurrency:
- failed descriptor.value undefined
- does not match worker scope

Screen.availHeight:
- failed descriptor.value undefined

Screen.availLeft:
- failed descriptor.value undefined

Screen.availTop:
- failed descriptor.value undefined

Screen.availWidth:
- failed descriptor.value undefined

Screen.colorDepth:
- failed descriptor.value undefined

Screen.pixelDepth:
- failed descriptor.value undefined

**Reply by abrahamjuliot** [reply]
[discussions/149](https://github.com/abrahamjuliot/creepjs/discussions/149) can be a helpful intro. Check out the articles near the end. I'm here if you have any questions.

**Reply by 190031994Harsha** [reply]
@abrahamjuliot Thankful for your reply sir..! .I am getting a trust score of 0%. I am a total beginner can someone tell what is this and how can I increase my trust score to the maximum. I am just scarping this website casually and making a report for educational purpose

**Reply by abrahamjuliot** [reply]
Are you using a library or custom script? I have no advice there, except to try without it. A higher trust rating can be achieved by following two steps.

Use a clean installation of the browser, then visit the site once daily for a few days.

Clean installation isn't necessary, but it might give you a better score.

**Reply by 190031994Harsha** [reply]
`const puppeteer = require('puppeteer');
const generatePDF = require('./generatePDF'); // Assuming this is a module you've created

async function scrapeAndGeneratePDF(loop) {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  // Set a random User-Agent string
  const userAgentList = ['Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edge/17.17134',
  "Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36",
  "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:31.0) Gecko/20100101 Firefox/31.0",
  "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36",
  "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.100 YaBrowser/16.11.1.673 Yowser/2.5 Safari/537.36",
  "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36",
  "Mozilla/5.0 (Windows NT 6.1; rv:52.0) Gecko/20100101 Firefox/52.0",
  "Mozilla/5.0 (Windows NT 6.1; rv:45.0) Gecko/20100101 Firefox/45.0",
  "Mozilla/5.0 (Windows NT 6.1; rv:38.0) Gecko/20100101 Firefox/38.0",
  "Mozilla/5.0 (Windows NT 6.1; rv:31.0) Gecko/20100101 Firefox/31.0",
  "Mozilla/5.0 (Windows NT 6.1; rv:24.0) Gecko/20100101 Firefox/24.0",
  "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 YaBrowser/17.6.0.1633 Yowser/2.5 Safari/537.36",
  "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.76 Safari/537.36 OPR/43.0.2442.806 (Edition Yx)",
  "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 YaBrowser/17.1.0.2034 Yowser/2.5 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.100 YaBrowser/16.11.1.673 Yowser/2.5 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36",
  "Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36 OPR/42.0.2393.94",
  "Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36",
  "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36",
  "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36",
  "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36",
  "Mozilla/5.0 (Linux; Android 6.0; thl T9 Build/MRA58K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.85 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 6.0.1; SM-G925I Build/MMB29K; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/56.0.2924.87 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 6.0.1; Redmi Note 3 Build/MMB29M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 5.1; Micromax Q334 Build/LMY47I) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 5.0; PowerFive Build/LRX21M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.4.4; MFLogin3T Build/KTU84P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.4.4; MFLogin3T Build/KTU84P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.135 Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.4.2; TZ707 Build/KOT49H) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.4.2; 9005X Build/KOT49H) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/30.0.0.0 Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.1.2; LG-E455 Build/JZO54K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.36",
  "Mozilla/5.0 (Linux; Android 4.1.2; LG-E455 Build/JZO54K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.116 YaBrowser/16.10.2.1487.00 Mobile Safari/537.36",
  "Mozilla/5.0 (iPad; CPU OS 5_0_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9A405 Safari/7534.48.3",
  "Mozilla/5.0 (Android 5.0; Mobile; rv:38.0) Gecko/20100101 Firefox/38.0",
  "Mozilla/5.0 () AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393",
  "Mozilla/5.0 (Linux; Android 13; SM-S906E) AppleWebKit/537.36 (KHTML, like Gecko) Soul/4.0 Chrome/108.0.5359.128 Mobile Safari/537.36"];
  const randomUserAgent = userAgentList[Math.floor(Math.random() * userAgentList.length)];
  //const randomUserAgent = userAgentList[userAgentList.length-1]
  await page.setUserAgent(randomUserAgent);

  console.log(randomUserAgent);
  await page.goto('https://abrahamjuliot.github.io/creepjs/');

  // Emulate human-like behavior (add delays between actions)
  await page.waitForTimeout(Math.random() * 2000 + 1000); // Random delay between 1-3 seconds
  await page.waitForTimeout(4000)
  // Scrape information using provided XPath expressions

  // Scrape Trust Percent
  const trustPercentElement = await page.$x('//div[starts-with(text(),"trust score")]//span[@class="unblurred"]');
  const trustPercent = await page.evaluate(element => element.textContent, trustPercentElement[0]);
  const trustPercent1 = trustPercent.trim(); // Assuming trustPercent is a string with leading/trailing spaces
  //console.log(trustPercent1);

  // Scrape Lies
  const liesElement = await page.$x('//div[starts-with(text(),"lies")]');
  const lies = await page.evaluate(element => element.textContent, liesElement[0]);
  //console.log(lies);

  // Process Lies (Remove unwanted characters)
  let txt = '';
  for (let i = 0; i < lies.length; i++) {
    if (lies[i] !== ')') {
      txt += lies[i];
    } else {
      txt += lies[i];
      break;
    }
  }
  //console.log(txt);

  // const browserele = await page.$x('//div[@class="ellipsis relative"]//span[@class="user-agent" and @xpath="1"]');
  // const browsername = await page.evaluate(element => element.textContent, browserele[0]);
  // console.log(browsername);

  // Process Lies Result
  let liesresult = '';
  if (txt !== 'lies (0): none') {
    const liescodeelement = await page.$x('//div[starts-with(text(),"lies")]//label[@class="modal-open-btn"]');
    const liescode = await page.evaluate(element => element.textContent, liescodeelement[0]);
    //console.log(liescode);

    const liesgradeelement = await page.$x('//div[starts-with(text(),"lies")]//span');
    const liesgrade = await page.evaluate(element => element.textContent, liesgradeelement[0]);
    //console.log(liesgrade);

    liesresult = `${txt}: ${liescode} ${liesgrade}`;
    //console.log(liesresult);
  } else {
    liesresult = txt;
    //console.log(liesresult);
  }

  // Scrape Bot
  const botElement = await page.$x('//div[starts-with(text(),"bot")]');
  const bot = await page.evaluate(element => element.textContent, botElement[0]);
  //console.log(bot);

  // Scrape Fingerprint
  const fingerprintElement = await page.$x('//div[contains(text(),"hidden fingerprint:")]//span');
  const fingerprint = await page.evaluate(element => element.textContent, fingerprintElement[0]);
  //console.log(fingerprint);

  const trustScoring = trustPercent1;

  // Create a unique directory for today's date
  generatePDF.createUniqueDirectory();

  // Generate PDF using scraped data
  generatePDF.generatePDF(trustScoring, liesresult, bot, fingerprint, loop);

  await browser.close();
}

module.exports = scrapeAndGeneratePDF;`

**Reply by 190031994Harsha** [reply]
this is the script i'm using

**Reply by abrahamjuliot** [reply]
This is a mystery to me. By chance, do you have DuckDuckGo or another extension installed in the browser?

**Reply by 190031994Harsha** [reply]
I have installed duck duck go for chrome only and this script randomly picks a browser when I run it even though if it is picking different browser the result is still the same

**Reply by abrahamjuliot** [reply]
It's likely DuckDuckGo. The lie pattern is very specific to the extension. It should go away if you turn off protections for the site. However, the extension is not necessarily a bad idea, and a low score for this pattern is exceptionally normal and common on the site.

**Reply by 190031994Harsha** [reply]
Then if it is normal how can I increase it ?
I read somewhere having it low is not a bad thing sometimes but still how can I increase it

**Reply by abrahamjuliot** [reply]
There is no feasible means to enhance the score, given the substantial number of mistrustful deviations from native code. It's basic for scripts to mimic known good patterns, and we can't give much trust to anything not native. 

As an alternative, you can await the crowd-blending score in the prediction section and prefer the higher score. If you have a 70% or higher crowd-blending score, then your browser is doing fairly well.

**Reply by 190031994Harsha** [reply]
even if I run the scripts now as my system data is already detected does the trust score and crowd blending scores be low?

**Reply by abrahamjuliot** [reply]
The crowd blending score can improve overtime. The only conditions required are 3 or more dates, same fingerprints, different IPs per date. However, the score per each prediction hash is discarded if the last visit ages 7 days.

The trust score factors in the crowd blending score but includes so much more, and is overall stricter and secretive.
